### PR TITLE
Skill - Resolve

### PR DESCRIPTION
**Pull request by JesterWizard**
When HP is < 50%,, SKL, SPD and STR * 1.5

### CODE REVIEW

File: Wizardry/Core/UnitStatusGetter/source/PowGetter.c [hunk 1]
```diff
@@ -86,6 +86,17 @@ int PowGetterSkills(int status, struct Unit * unit)
             status += 5;
 #endif
     }
+    
+#if (defined(SID_Resolve) && (SID_Resolve < MAX_SKILL_NUM)) 
+        if (SkillTester(unit, SID_Resolve))
+        {
+            if (((GetUnitMaxHp(unit) % 2 == 0) && (GetUnitCurrentHp(unit) < GetUnitMaxHp(unit)/2)) ||
```
Comment (MokhaLeee) [inline Wizardry/Core/UnitStatusGetter/source/PowGetter.c:93]: why not directly use Multiplication？

File: Wizardry/Core/UnitStatusGetter/source/PowGetter.c [hunk 2]
```diff
@@ -86,6 +86,17 @@ int PowGetterSkills(int status, struct Unit * unit)
...
+                ((GetUnitMaxHp(unit) % 2 != 0) && (GetUnitCurrentHp(unit) -1 < GetUnitMaxHp(unit)/2)))
+                {
+                    status += status/2;
```
Comment (MokhaLeee) [inline Wizardry/Core/UnitStatusGetter/source/PowGetter.c:96]: add space for "/"， Please be consistent with the code style of other parts.

Comment (JesterWizard) [reply Wizardry/Core/UnitStatusGetter/source/PowGetter.c]: Done

File: Wizardry/Core/UnitStatusGetter/source/PowGetter.c (hunk 1 above)
Comment (JesterWizard) [reply Wizardry/Core/UnitStatusGetter/source/PowGetter.c]: Done

File: Wizardry/Core/UnitStatusGetter/source/SklGetter.c [hunk 3]
```diff
@@ -70,5 +70,15 @@ int SklGetterSkills(int status, struct Unit * unit)
 #endif
     }
 
+#if (defined(SID_Resolve) && (SID_Resolve < MAX_SKILL_NUM)) 
+        if (SkillTester(unit, SID_Resolve))
+        {
+            if (((GetUnitMaxHp(unit) % 2 == 0) && (GetUnitCurrentHp(unit) < GetUnitMaxHp(unit) / 2)) ||
+                ((GetUnitMaxHp(unit) % 2 != 0) && (GetUnitCurrentHp(unit) -1 < GetUnitMaxHp(unit) / 2)))
+                {
+                    status *= 1.5;
```
Comment (MokhaLeee) [inline Wizardry/Core/UnitStatusGetter/source/SklGetter.c:79]: WTF... Where does the floating-point number come from in embedded systems?

File: Wizardry/Core/UnitStatusGetter/source/SklGetter.c [hunk 4]
```diff
@@ -70,5 +70,15 @@ int SklGetterSkills(int status, struct Unit * unit)
 #endif
     }
 
+#if (defined(SID_Resolve) && (SID_Resolve < MAX_SKILL_NUM)) 
+        if (SkillTester(unit, SID_Resolve))
+        {
+            if (((GetUnitMaxHp(unit) % 2 == 0) && (GetUnitCurrentHp(unit) < GetUnitMaxHp(unit) / 2)) ||
+                ((GetUnitMaxHp(unit) % 2 != 0) && (GetUnitCurrentHp(unit) -1 < GetUnitMaxHp(unit) / 2)))
```
Comment (MokhaLeee) [inline Wizardry/Core/UnitStatusGetter/source/SklGetter.c:77]: Replace these division calculations with multiplication

**Comment by MokhaLeee** [comment]
The calculation is not correct, I decide diectly fix it after merging

**Comment by MokhaLeee** [comment]
https://github.com/MokhaLeee/fe8u-cskillsys-kernel/commit/3b86431eba63aca911aa90122fed133dd347c031

**Pull request merged**